File: code/smv-debugger/src/main/java/org/fbme/smvDebugger/commons/OSUtils.java
package org.fbme.smvDebugger.commons;

import java.util.EnumSet;

public final class OSUtils {
  private static final EnumSet<OS> SUPPORTED_OS = EnumSet.<OS>of(OS.MAC_OS);

  public static OS getOS() {
    final String osName = System.getProperty("os.name").toLowerCase();
    if (osName.contains("mac os")) {
      return OS.MAC_OS;
    }
    if (osName.contains("windows")) {
      return OS.WINDOWS;
    }
    return OS.UNKNOWN;
  }

  public static boolean isSupportedOS(final OS os) {
    return SUPPORTED_OS.contains(os);
  }
}

File: code/smv-debugger/src/main/kotlin/org/fbme/smvDebugger/commons/OSUtils.kt
package org.fbme.smvDebugger.commons

import java.util.*

object OSUtils {
    private val SUPPORTED_OS = EnumSet.of(OS.MAC_OS)
    val oS: OS
        get() {
            val osName = System.getProperty("os.name").lowercase(Locale.getDefault())
            if (osName.contains("mac os")) {
                return OS.MAC_OS
            }
            return if (osName.contains("windows")) {
                OS.WINDOWS
            } else OS.UNKNOWN
        }

    fun isSupportedOS(os: OS): Boolean {
        return SUPPORTED_OS.contains(os)
    }
}
